### DISCUSSION TITLE
Windows Package Manager 1.7.2782-preview

### DISCUSSION THREAD

**Discussion by ryfu-msft** [microsoft/winget-cli#3736] [Releases]
This is the fourth development build after the Windows Package Manager 1.6 build for Windows 10 (1809+) and Windows 11. This build will be released to Windows Insider Dev builds and Windows Package Manager Insiders. 

Experimental features are enabled in this release. Run `winget features` to see which experimental features are enabled or disabled. Add the following to your settings (`winget settings`) file to enable the experimental features.

```JSON
    "experimentalFeatures": {
	  "directMSI": true
          "windowsFeature": true
    },
```

Windows Package Manager also includes [Winget configuration](https://learn.microsoft.com/windows/package-manager/configuration/), which automatically handles the setup and configuration requirements for an ideal development environment on your Windows machine. WinGet configuration file helps with installing and managing software packages, applications, programming languages, frameworks, tools, or settings necessary for a project. 

Check out our [session](https://aka.ms/BRK280H) at Microsoft Build to learn how to get your machine to a ready-to-code state.

A prerelease version of the [Microsoft.WinGet.Client PowerShell module](https://www.powershellgallery.com/packages/Microsoft.WinGet.Client/0.2.1) has been published to the PowerShell Gallery and will no longer be included as a release asset. To install the latest version of the PowerShell module, run the following command in PowerShell 7+. 

```pwsh
Install-Module -Name Microsoft.WinGet.Client
```
> The PowerShell module requires *App Installer ([winget](https://aka.ms/getwinget))* to be installed. The `Repair-WinGetPackageManager` cmdlet (work in progress) is designed to install or repair *App Installer*.

## What's Changed
* Platform source should not be blocked by policy check by @yao-msft in https://github.com/microsoft/winget-cli/pull/3725
* Update docs for upgrade command by @KK-Designs in https://github.com/microsoft/winget-cli/pull/3639
* Honor 429 Retry-After by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3718

## New Contributors
* @KK-Designs made their first contribution in https://github.com/microsoft/winget-cli/pull/3639

**Full Changelog**: https://github.com/microsoft/winget-cli/compare/v1.7.2722-preview...v1.7.2782-preview

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/winget-cli/releases/tag/v1.7.2782-preview'>Windows Package Manager 1.7.2782-preview</a>.</em>

**Comment by mavaddat** [comment]
To enable the experimental features, we need the syntactically correct JSONC settings below:

```json
    "experimentalFeatures": {
	  "directMSI": true,
          "windowsFeature": true
    },
```

The difference:

```diff
2c2
< 	  "directMSI": true
---
> 	  "directMSI": true,
```
